osier/technology.py

```diff
@@ -319,7 +346,152 @@ def variable_cost_ts(self, size):
+        then the technology may ramp up its power level by 50% per unit time.
+        The default is 1.0 (i.e. there is no constraint on ramping up).
+
+    ramp_down_rate : float, :class:`unyt_quantity`
+        The rate at which a technology can decrease its power, expressed as
+        a percentage of its capacity. For example, if `ramp_down_rate` equals 0.5,
+        then the technology may ramp down its power level by 50% per unit time.
+        The default is 1.0 (i.e. there is no constraint on ramping down).
+
+
+    Notes
+    -----
+    It is common for a ramping technology to have different ramp up and ramp
+    down rates. Consider a light-water nuclear reactor that can quickly reduce
+    its power level by inserting control rods, but must wait much longer to
+    increase its power by the same amount due to a build up of neutron
+    absorbing isotopes.
+    """
+
+    def __init__(
+            self,
+            technology_name,
+            technology_type='production',
+            technology_category='ramping',
+            dispatchable=True,
+            renewable=False,
+            capital_cost=0,
+            om_cost_fixed=0,
+            om_cost_variable=0,
+            fuel_cost=0,
+            fuel_type=None,
+            capacity=0,
+            capacity_factor=1.0,
+            efficiency=1.0,
+            default_power_units=MW,
+            default_time_units=hr,
+            default_energy_units=None,
+            ramp_up_rate=1.0 * hr**-1,
+            ramp_down_rate=1.0 * hr**-1) -> None:
+        super().__init__(
+            technology_name,
+            technology_type,
+            technology_category,
+            dispatchable,
+            renewable,
+            capital_cost,
+            om_cost_fixed,
+            om_cost_variable,
+            fuel_cost,
+            fuel_type,
+            capacity,
+            capacity_factor,
+            efficiency,
+            default_power_units,
+            default_time_units,
+            default_energy_units)
+
+        self.ramp_up_rate = _validate_quantity(ramp_up_rate,
+                                               dimension='spec_time')
+        self.ramp_down_rate = _validate_quantity(ramp_down_rate,
+                                                 dimension='spec_time')
+
+    @property
+    def ramp_up(self):
+        return (
+            self.capacity *
+            self.ramp_up_rate).to(
+            self.unit_power *
+            self.unit_time**-1
+        )
+
+    @property
+    def ramp_down(self):
+        return (
+            self.capacity *
+            self.ramp_down_rate).to(
+            self.unit_power *
+            self.unit_time**-1
+        )
+
+
+class ThermalTechnology(RampingTechnology):
+    """
+    The :class:`ThermalTechnology` class extends the :class:`RampingTechnology`
+    class by adding a heat rate.
+
+    Parameters
+    ----------
+    heat_rate : int, float
+        The heat rate of a given technology.
```

Comment:
Can you expand on what you mean by heat? I don't think this is precise enough, but maybe the way you are using the term is commonplace in the electrical utility field?

Reply:
Yes, it's an extremely common way to denote efficiency (but it only applies to thermal plants!). It's the amount of "heat" (energy) required to generate 1 kWh of electricity. EIA and NREL both use this term.